doc/admin-guide/plugins/filter_body.en.rst

```diff
@@ -0,0 +1,330 @@
+- Configurable HTTP methods to match (GET, POST, PUT, etc.)
+
+Installation
+============
+
+The ``filter_body`` plugin is an experimental plugin. To build it, pass
+``-DENABLE_FILTER_BODY=ON`` to ``cmake`` when configuring the build::
+
+    cmake -DENABLE_FILTER_BODY=ON ...
+
+Configuration
+=============
+
+The plugin is configured as a remap plugin with a YAML configuration file::
+
+    map http://example.com/ http://origin.example.com/ @plugin=filter_body.so @pparam=filter_body.yaml
+
+The configuration file path can be relative to the |TS| configuration directory
+or an absolute path.
+
+Configuration File Format
+-------------------------
+
+The configuration file uses YAML format with a list of rules::
+
+    rules:
+      - name: rule_name
+        direction: request|response
+        methods:
+          - POST
+          - PUT
+        max_content_length: 1048576
+        headers:
+          - name: Content-Type
+            patterns:
+              - "application/xml"
+              - "text/xml"
+        body_patterns:
+          - "<!ENTITY"
+          - "<!DOCTYPE"
+        action:
+          - log
+          - block
+        add_header_name: X-Security-Match
+        add_header_value: "rule_name"
+
+Rule Options
+------------
+
+``name`` (required)
+    A unique name for the rule. Used in log messages when the rule matches.
+
+``direction`` (required)
+    Specifies whether to inspect request or response bodies.
+    Valid values: ``request``, ``response``
+
+``methods`` (optional)
+    List of HTTP methods to match. If not specified, all methods are matched.
+    Example: ``[GET, POST, PUT]``
+
+``max_content_length`` (optional)
+    Maximum content length in bytes for body inspection. Bodies larger than
+    this value will not be inspected. If set to 0 or not specified, all bodies
+    are inspected regardless of size.
+
+``headers`` (optional)
+    List of header conditions that must all match (AND logic) for body
+    inspection to occur. Each header can have multiple patterns (OR logic
+    within a single header).
+
+    - ``name``: Header name (case-insensitive matching)
+    - ``patterns``: List of patterns to match against the header value
+
+``body_patterns`` (required)
+    List of patterns to search for in the body content. Pattern matching is
+    case-sensitive. If any pattern matches, the configured actions are executed.
+
+``action`` (optional)
+    List of actions to take when a pattern matches. Default is ``[log]``.
+    Valid values:
+
+    - ``log``: Log the match to the debug log (requires debug tags enabled)
+    - ``block``: Block the request/response with a 403 Forbidden status
+    - ``add_header``: Add a custom header to the request/response
+
+``add_header_name`` (optional)
+    Name of the header to add when ``add_header`` action is configured.
+
+``add_header_value`` (optional)
+    Value of the header to add. Defaults to the rule name if not specified.
```

Comment:
Indicate that `<rule_name>` is a directive indicating that the rules:name node value should be used for the http header field value.